lib/gis_robot_suite/vector_normalizer.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module GisRobotSuite
+  class VectorNormalizer
+    def initialize(logger:, bare_druid:, rootdir:)
+      @logger = logger
+      @bare_druid = bare_druid
+      @rootdir = rootdir
+    end
+
+    def with_normalized
+      yield normalize
+    ensure
+      cleanup
+    end
+
+    # @return [String] the path to the normalized vector
+    def normalize
+      FileUtils.mkdir_p tmpdir
+
+      normalize_shp
+      normalize_prj
+
+      tmpdir
+    end
+
+    def cleanup
+      logger.debug "Cleaning: #{tmpdir}"
+      FileUtils.rm_rf tmpdir
+    end
+
+    def run_shp2pgsql(projection, encoding, shpfn, schema, sqlfn, errfn)
+      # XXX: Perhaps put the .sql data into the content directory as .zip for derivative
+      # XXX: -G for the geography column causes some issues with GeoServer
+      cmd = "shp2pgsql -s #{projection} -d -D -I -W #{encoding} " \
+            "'#{shpfn}' #{schema}.#{bare_druid} " \
+            "> '#{sqlfn}' 2> '#{errfn}'"
+      logger.debug "Running: #{cmd}"
+      success = system(cmd, exception: true)
+      raise "normalize-vector: #{bare_druid} cannot convert Shapefile to PostGIS: #{File.open(errfn).readlines}" unless success
+      raise "normalize-vector: #{bare_druid} shp2pgsql generated no SQL?" unless File.size?(sqlfn)
+    end
```

Comment:
had to make this method public so it can be called by the `LoadVector` robot correctly